src/widgets/sparkline.rs

```diff
@@ -92,8 +117,8 @@ impl<'a> Sparkline<'a> {
     /// # }
     /// ```
     #[must_use = "method moves the value of self and returns the modified value"]
-    pub const fn data(mut self, data: &'a [u64]) -> Self {
-        self.data = data;
+    pub fn data(mut self, data: impl Into<SparklineData>) -> Self {
+        self.data = data.into();
```

Comment:
Added.  

I've split `SparklineData` into `SparkData(Vec<SparkItem>)`.  Happy to bike shed on names.

`Sparks` / `Spark`?
`SparkLineData` / `SparkLineItem`?
`Data` / `Item`?

Reply:
I've managed to eliminate `SparkData`, so now the signature of data is reduced to:

```rust
    pub fn data<T>(mut self, data: T) -> Self
    where
        T: IntoIterator,
        T::Item: Into<SparkItem>
```

This should, I suspect, also reduce issues with type inference.

Reply:
Added to `BREAKING-CHANGES.md`:

```
- [v0.29.0](#v0290)
  - `Sparkline::data` is no longer declared as `const`
  - `Sparkline::data` now takes `IntoIterator<Item = SparklineBar>` instead of `&[u64]`
```

I'm unclear on what to add to the documentation based on your example, can you clarify?  Perhaps starting with an example where inference will be an issue and then showing the recommended solution.

Reply:
The broken code is anything where you'd pass a value that can be converted into u64.
A simpler example of this is:
```rust
let value = 1u8;
Sparkline::default().data(&[value.into()]);
```
The fix is:
```rust
Sparkline::default().data(&[u64::from(value)]);
```

Reply:
Understood.  I added the example and some commentary to the `BREAKING-CHANGES.md` details.